app/ota_boot.py

```diff
@@ -1,78 +1,203 @@
 #!/usr/bin/env python3
-
-import os
+from abc import ABC, abstractmethod
+from pathlib import Path
 import shutil
 
-import ota_status
-import grub_control
+from ota_status import OtaStatus
+from grub_control import GrubCtl
+from constants import OtaBootStatusString, OtaStatusString
+from exceptions import OtaBootError
+import configs as cfg
 
 from logging import getLogger, INFO, DEBUG
 
 logger = getLogger(__name__)
-logger.setLevel(INFO)
+logger.setLevel(cfg.LOG_LEVEL_TABLE.get(__name__, cfg.DEFAULT_LOG_LEVEL))
+
+
+class OtaBootInterface(ABC):
+    """
+    OtaBoot interface for implementing OtaBoot
+    """
+
+    # alias
+    # stage 1: check
+    check_passed, check_failed = True, False
+    # stage 2: finalize
+    finalize_succeeded, finalize_failed = True, False
+    # return value
+    bypass_finalization_check_passed, bypass_finalization_check_failed, bypass_check = (
+        "no_f_c_p",
+        "no_f_c_f",
+        "no_c",
```

Comment:
I understand what is this now.
"no_f_c_p" should be "bypass_finalization_check_passed",
"no_f_c_f" should be "bypass_finalization_check_failed", and
"no_c" should be "bypass_check", to understand easily.

Reply:
Yes, because we don't use the value of these variables, so I just use the abbreviation of these long sentences.

Reply:
Will be updated with an separate enum class.

Reply:
Replace by enum in commit 68b869a.
Now these variables are used as pure enums without caring about the concrete values.